includes/templates/template_default/templates/tpl_modules_shipping_estimator.php

```diff
@@ -98,14 +98,17 @@
        <th scope="col" id="seTotalHeading"><?php echo CART_SHIPPING_METHOD_RATES; ?></th>
      </tr>
 <?php
+      if (empty($extra)) {
+        $extra = '';
+      }
       for ($i=0, $n=sizeof($quotes); $i<$n; $i++) {
-        if(sizeof($quotes[$i]['methods'])==1){
+        if(isset($quotes[$i]['id']) && sizeof($quotes[$i]['methods'])==1 && isset($quotes[$i]['methods'][0]['id'])){

```

Comment:
Should `$thisquoteid` be initialized here, similar to what you added later?

Reply:
Depends on overall code structure goal.  The else of this `if` statement is where an initialization of `$thisquoteid` was added.  The first step within this first `if` statement was to make an assignment to the variable `$thisquoteid`.  

So, if there is a goal to perhaps follow some other programming languages where every variable that is to be used is to be declared at the onset of the function and possibly provided an "initial" or default value, then yes would suggest initializing `$thisquoteid` to an empty string early on and/or possibly on each `for` loop. Probably would be more important to have the value `$thisquoteid` defined to something if there were any other location later in the code that needed to use the variable.

So, even though I think by this point the code has already been incorporated, doesn't mean that further changes aren't possible/necessary. :)

Reply:
While there's no specific goal to pre-initialize every possible variable, PHP's embracing of strictness causes the need for all these `isset` checks if the variable is going to be passed or its contents evaluated.

I had observed your pattern, and wondered if you had intentionally skipped initializing it in the first instance when you had initialized it in the second.